tests/ha/test_control_pod_restart.py

```diff
@@ -0,0 +1,417 @@
+from http import HTTPStatus
+
+# Global Constants
+LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=R0902
+# pylint: disable=R0904
+class TestControlPodRestart:
+    """
+    Test suite for Control Pod Restart
+    """
+
+    @classmethod
+    def setup_class(cls):
+        """
+        Setup operations for the test file.
+        """
+        LOGGER.info("STARTED: Setup Module operations.")
+        cls.csm_user = CMN_CFG["csm"]["csm_admin_user"]["username"]
+        cls.csm_passwd = CMN_CFG["csm"]["csm_admin_user"]["password"]
+        cls.num_nodes = len(CMN_CFG["nodes"])
+        cls.username = []
+        cls.password = []
+        cls.host_master_list = []
+        cls.node_master_list = []
+        cls.hlth_master_list = []
+        cls.host_worker_list = []
+        cls.node_worker_list = []
+        cls.ha_obj = HAK8s()
+        cls.deploy_lc_obj = ProvDeployK8sCortxLib()
+        cls.s3_clean = cls.test_prefix = cls.random_time = None
+        cls.s3acc_name = cls.s3acc_email = cls.bucket_name = cls.object_name = None
+        cls.restore_pod = cls.deployment_backup = cls.deployment_name = cls.restore_method = None
+        cls.restore_node = cls.node_name = cls.deploy = None
+        cls.restore_ip = cls.node_iface = cls.new_worker_obj = cls.node_ip = None
+        cls.mgnt_ops = ManagementOPs()
+        cls.system_random = secrets.SystemRandom()
+        cls.motr_obj = MotrCoreK8s()
+        cls.rest_iam_user = RestIamUser()
+
+        for node in range(cls.num_nodes):
+            cls.host = CMN_CFG["nodes"][node]["hostname"]
+            cls.username.append(CMN_CFG["nodes"][node]["username"])
+            cls.password.append(CMN_CFG["nodes"][node]["password"])
+            if CMN_CFG["nodes"][node]["node_type"] == "master":
+                cls.host_master_list.append(cls.host)
+                cls.node_master_list.append(LogicalNode(hostname=cls.host,
+                                                        username=cls.username[node],
+                                                        password=cls.password[node]))
+                cls.hlth_master_list.append(Health(hostname=cls.host,
+                                                   username=cls.username[node],
+                                                   password=cls.password[node]))
+            else:
+                cls.host_worker_list.append(cls.host)
+                cls.node_worker_list.append(LogicalNode(hostname=cls.host,
+                                                        username=cls.username[node],
+                                                        password=cls.password[node]))
+
+        cls.rest_obj = S3AccountOperations()
+        cls.s3_mp_test_obj = S3MultipartTestLib(endpoint_url=S3_CFG["s3_url"])
+        cls.test_file = "ha-mp_obj"
+        cls.test_dir_path = os.path.join(TEST_DATA_FOLDER, "HATestMultipartUpload")
+        control_pods = cls.node_master_list[0].get_pods_node_fqdn(const.CONTROL_POD_NAME_PREFIX)
```

Comment:
So, we need not required to take backup after each test ? ie. Do we need it in setup_method ?

Reply:
Not needed after each test.

Reply:
If you are failing over, won't this chnage? whre are we using it exactly?

Reply:
In teardown, to change pod node to its original pod, with original PV settings.

Reply:
We already have this in setup_method.
But in setup_method we may get different params than that of original so added this in setup_class as well, and using same in teardown for reverting changes.